va/va.h

```diff
@@ -129,6 +129,96 @@ extern "C" {
  * 	- \ref api_fei
  * 	- \ref api_fei_h264
  * 	- \ref api_fei_hevc
+ * 
+ * \section threading Multithreading Guide
+ *
+ * It is assumed that all VAAPI functions are thread-safe. As most of the
+ * functions are implemented by the VAAPI driver or backend, it is the
+ * responsibility of the driver to guarantee thread-safty of the backend
+ * functions. It is assumed that none of the VAAPI functions will be called from
+ * signal handlers.
+ *
+ * Thread-safty in this context means that when VAAPI is being called by multiple
+ * concurrent threads, it will not crash or hang the OS, and VAAPI internal
+ * states will not be corrupted. When multiple threads are operating on the same
+ * VAAPI objects, it it the application's responsibility to synchronize these
+ * operations in order to generate the expected results.
+ *
```

Comment:
These two paragraphs are confusing, because it says that `VAAPI functions are thread-safe` but then `it is the responsibility of the driver to guarantee thread-safty [sic]`, so: please rewrite the sections to reflect what you mean, e.g.: `VA-API internal objects are thread safe, but VA-API does not have nor impose or provide any guarantees of thread-safety on the underlying driver or backend, nor on any application-managed VA-API objects, including in particular VAContext `.

Reply:
Thanks for the comments and patch will be updated.